File: Leetcode Problems/125.py
from typing import List


class Solution:
    def isPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s) - 1

        while i < j:
            if not s[i].isalnum():
                i += 1
                continue
            
            if not s[j].isalnum():
                j -= 1
                continue
            
            if s[i].lower() != s[j].lower():
                return False
            
            i += 1  
            j -= 1
        
        return True
    
if __name__ == '__main__':
    s1 = "aahhaa"
    s2 = "byubvyuifvb"
    
    solution = Solution()
    print(solution.isPalindrome(s1))
    print(solution.isPalindrome(s2))

File: Leetcode Problems/125.cpp
#include<iostream>
#include<string>
using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        int i = 0;
        int j = s.size() - 1;

        while(i < j) {
            if(!isalnum(s[i])) {
                i++;
            }

            if(!isalnum(s[j])) {
                j--;
            }
            
            if(tolower(s[i]) != tolower(s[j])) {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
};

int main() {
    string s1 = "aahhaa";
    string s2 = "aruiyfvyubvr";

    Solution obj;
    cout<<obj.isPalindrome(s1)<<endl;
    cout<<obj.isPalindrome(s2)<<endl;

    return 0;
}